app/components/markdown/markdown_link/markdown_link.tsx

```diff
@@ -123,10 +123,14 @@ const MarkdownLink = ({children, experimentalNormalizeMarkdownLinks, href, siteU
                             leftIcon='content-copy'
                             onPress={() => {
                                 dismissBottomSheet();
-                                Clipboard.setString(href);
+                                const cleanHref = href.replace(/^mailto:/, '');
+                                Clipboard.setString(cleanHref);
                             }}
                             testID='at_mention.bottom_sheet.copy_url'
-                            text={intl.formatMessage({id: 'mobile.markdown.link.copy_url', defaultMessage: 'Copy URL'})}
+                            text={intl.formatMessage({
+                                id: isEmail(href.replace(/^mailto:/, '')) ? 'mobile.markdown.link.copy_email' : 'mobile.markdown.link.copy_url',
+                                defaultMessage: isEmail(href.replace(/^mailto:/, '')) ? 'Copy Email Address' : 'Copy URL',
+                            })}
```

Comment:
Due to how our intl tooling works, this will mess with the string extraction. In order to make this more readable, use a new variable (defined outside of the component) like this:

```
const messages = defineMessages({
    copyEmail: {
        id: 'mobile.markdown.link.copy_email',
        defaultMessage: 'Copy Email Address',
    },
    copyURL: {...}
});
```

Also, let's store the cleanHref in a variable outside `renderContent`, so we can reuse it. That way the next check looks cleaner too.

Finally, you can just do:
```
intl.formatMessage(isEmail(cleanRef) ? messages.copyEmail : message.copyURL)
```